### PR TITLE
Added missing tests for AnonymousUser methods and properties.

### PR DESCRIPTION
**Pull request by atombrella**
It seems better to use `SimpleTestCase` for these tests; as a result I moved the existing properties test into this new class. An assert for `get_group_permissions` was added to the properties-test.

### CODE REVIEW

File: tests/auth_tests/test_basic.py [hunk 1]
```diff
@@ -121,6 +107,63 @@ def test_user_verbose_names_translatable(self):
             self.assertEqual(User._meta.verbose_name_plural, 'usuarios')
 
 
+class AnonymousUserTests(SimpleTestCase):
+    def setUp(self):
+        self.no_db_repr_msg = (
+            "Django doesn't provide a DB representation for AnonymousUser."
```
Comment (timgraham) [inline tests/auth_tests/test_basic.py]: This can be a class attribute.

File: tests/auth_tests/test_basic.py [hunk 2]
```diff
@@ -121,6 +107,63 @@ def test_user_verbose_names_translatable(self):
...
+        )
+
+    def test_properties(self):
+        """Check the properties of the anonymous user."""
+        a = AnonymousUser()
+        self.assertIsNone(a.pk)
+        self.assertEqual(a.username, '')
+        self.assertEqual(a.get_username(), '')
+        self.assertTrue(a.is_anonymous)
```
Comment (timgraham) [inline tests/auth_tests/test_basic.py]: When moving, please use `assertIs()` instead.

File: tests/auth_tests/test_basic.py [hunk 3]
```diff
@@ -121,6 +107,63 @@ def test_user_verbose_names_translatable(self):
...
 
 
+class AnonymousUserTests(SimpleTestCase):
+    def setUp(self):
+        self.no_db_repr_msg = (
+            "Django doesn't provide a DB representation for AnonymousUser."
+        )
+
+    def test_properties(self):
+        """Check the properties of the anonymous user."""
```
Comment (timgraham) [inline tests/auth_tests/test_basic.py]: Can be omitted.

File: tests/auth_tests/test_basic.py [hunk 4]
```diff
@@ -121,6 +107,63 @@ def test_user_verbose_names_translatable(self):
...
+        b = AnonymousUser()
+        c = User('super', 'super@example.com', 'super')
+        self.assertEqual(a, b)
+        self.assertNotEqual(a, c)
+
+    def test_hash(self):
+        a = AnonymousUser()
+        self.assertEqual(hash(a), 1)
+
+    def test_delete_not_implemented(self):
```
Comment (timgraham) [inline tests/auth_tests/test_basic.py]: I think `test_delete` is sufficient.

**Comment by atombrella** [comment]
@timgraham Can you see if this is what you wanted?

**Comment by timgraham** [comment]
Oops, we were both editing at the same time. Yes, that's it. I'm going to push my version as I also move the tests to `test_models.py` and made a few other cosmetic edits. If you feel like doing more cleanup, the rest of `test_basic.py` could be a candidate for better organization.

**Comment by atombrella** [comment]
@timgraham Thanks. I missed the "separate commits" comment initially. I'm fine with the content of the PR now.

**Pull request merged**